### ISSUE TITLE
Looking to integrate CloudEvents with gRPC - please help

### ISSUE THREAD

**Issue by VenkateshSrini** [cloudevents/sdk-csharp#248]
Hi,

I looking to integrate CloudEvents and gRPC. gRPC will be the standard protocol format while the CloudEvents will be the standard message structure. In the CloudEvents go SDK I see that the [proto](https://github.com/cloudevents/sdk-go/tree/main/binding/format/protobuf/v2/pb)definition is available. Is there something available like that in the C# SDK also? Also, some samples in this regard will be of great help.

**Comment by jskeet** [comment]
Yes, we have [CloudNative.CloudEvents.Protobuf](https://github.com/cloudevents/sdk-csharp/tree/main/src/CloudNative.CloudEvents.Protobuf) - that uses a slightly different proto definition from the Go SDK, as it uses the one [in the CloudEvents spec repo](https://github.com/cloudevents/spec/blob/main/cloudevents/formats/cloudevents.proto). I haven't checked in detail, but my hope is that the only significant difference is the protobuf package declaration. It also looks like the Go SDK proto doesn't include the `CloudEventBatch` message.

There are no samples for this I'm afraid - but I'd recommend looking at the [tests](https://github.com/cloudevents/sdk-csharp/tree/main/test/CloudNative.CloudEvents.UnitTests/Protobuf).

**Comment by VenkateshSrini** [comment]
@jskeet please let me know what import statement should I use in my proto so that I can consume cloud events and return Cloudevents in. Proto file

**Comment by jskeet** [comment]
You'd need to clone the .proto file from the spec repo (see earlier link) first, and import that.

> Is there a way I can get. Proto file from generated code

No, you need the original .proto file.

**Comment by VenkateshSrini** [comment]
@jskeet thank you. Just to confirm is this the link that you are referring https://github.com/cloudevents/spec/blob/main/cloudevents/formats/cloudevents.proto

**Comment by jskeet** [comment]
Yes, exactly.
